File: src/C#/CreationalPatterns/Singleton/VotingMachine/VotingMachineV1.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace VotingMachine
{
    public sealed class VotingMachineV1
    {
        private int totalVotes = 0;

        private static VotingMachineV1 votingMachineInstance;

        private VotingMachineV1()
        {

        }
        public int TotalVotes => totalVotes;

        public static VotingMachineV1 VotingMachineInstance
        {
            get
            {
                votingMachineInstance ??= new VotingMachineV1();

                return votingMachineInstance;
            }
        }

        public void RegisterVote()
        {
            totalVotes++;
            Console.WriteLine($"Registered Vote #{totalVotes}");
        }
    }
}

File: src/Java/CreationalPatterns/Singleton/VotingMachine/src/VotingMachineV1.java
public final class VotingMachineV1 {
    private int totalVotes = 0;

    private static VotingMachineV1 votingMachineInstance;

    private VotingMachineV1() {

    }

    public int getTotalVotes() {
        return totalVotes;
    }

    public static VotingMachineV1 getVotingMachineInstance() {
        if (votingMachineInstance == null) {
            votingMachineInstance = new VotingMachineV1();
        }

        return votingMachineInstance;
    }

    public void registerVote() {
        totalVotes++;
        System.out.println("Registered Vote #" + totalVotes);
    }
}
